src/support.rs

```diff
@@ -297,6 +324,40 @@ impl<T: Shared> Borrow<T> for SharedRef<T> {
   }
 }
 
+#[track_caller]
+fn assert_shared_ptr_use_count_eq<P, T: Shared>(
+  shared_ptr: &SharedPtrBase<T>,
+  expected: usize,
+) {
+  let mut actual = T::use_count(shared_ptr);
+  let ok = match long::try_from(expected) {
+    Err(_) => false, // Non-`long` value can never match actual use count.
+    Ok(expected) if actual == expected => true, // Fast path.
+    Ok(expected) => {
+      pub const RETRY_TIMEOUT: Duration = Duration::from_secs(1);
+      let start = Instant::now();
+      loop {
+        yield_now();
+        actual = T::use_count(shared_ptr);
+        if actual == expected {
+          break true;
+        } else if start.elapsed() > RETRY_TIMEOUT {
+          break false;
+        }
+      }
```

Comment:
I don't follow what's happening here? Is this a one second hot loop? Why? (Maybe add a comment?)